app/lib/primer/classify/functional_colors.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Primer
+  class Classify
+    # https://primer-css-git-mkt-color-modes-docs-primer.vercel.app/css/support/v16-migration
+    class FunctionalColors
+      FUNCTIONAL_COLOR_REGEX = /(primary|secondary|tertiary|link|success|warning|danger|info|inverse|text_white)/.freeze
+
+      TEXT_COLOR_MAPPINGS = {
+        gray_dark: :text_primary,
+        gray: :text_secondary,
+        gray_light: :text_tertiary,
+        blue: :text_link,
+        green: :text_success,
+        yellow: :text_warning,
+        red: :text_danger,
+        white: :text_white,
+        # still unsure what will happen with these colors
+        black: nil,
+        orange: nil,
+        orange_light: nil,
+        purple: nil,
+        pink: nil,
+        inherit: nil
+      }.freeze
+
+      class << self
+        def text_color(val)
+          # the value is a functional color
+          return "color-#{val.to_s.dasherize}" if FUNCTIONAL_COLOR_REGEX.match?(val) || ends_with_number?(val)
+          # if the app still allows non functional colors
+          return "text-#{val.to_s.dasherize}" unless force_functional_colors?
+
+          if TEXT_COLOR_MAPPINGS.key?(val)
+            functional_color = TEXT_COLOR_MAPPINGS[val]
+            # colors without functional mapping stay the same
+            return "text-#{val.to_s.dasherize}" if functional_color.blank?
+
+            ActiveSupport::Deprecation.warn("Color #{val} is deprecated. Please use #{functional_color} instead.") unless Rails.env.production?
+
+            return text_color(functional_color)
+          end
+
+          raise ArgumentError, "Color #{val} does not exist."
```

Comment:
We haven't raised before when given invalid values. Should we log a warning instead?

Reply:
it will only reach here if the user opted in to `force_functional_colors`, so I think we are fine raising

Reply:
I share Blake's concern. Might we only raise this warning if we aren't in production?

Reply:
done!